ts/sagas/wallet/pagopaApis.ts

```diff
@@ -360,6 +366,86 @@ export function* updateWalletPspRequestHandler(
   }
 }
 
+/**
+ * delete all those payment methods that have the specified function enabled
+ * @param pagoPaClient
+ * @param pmSessionManager
+ * @param action
+ */
+export function* deleteAllPaymentMethodsByFunctionRequestHandler(
+  pagoPaClient: PaymentManagerClient,
+  pmSessionManager: SessionManager<PaymentManagerToken>,
+  action: ActionType<typeof deleteAllPaymentMethodsByFunction>
+): Generator<Effect, void, any> {
+  const deleteAllByFunctionApi = pagoPaClient.deleteAllPaymentMethodsByFunction(
+    action.payload as string
+  );
+  const deleteAllByFunctionApiWithRefresh = pmSessionManager.withRefresh(
+    deleteAllByFunctionApi
+  );
+
+  try {
+    const deleteResponse: SagaCallReturnType<typeof deleteAllByFunctionApiWithRefresh> = yield call(
+      deleteAllByFunctionApiWithRefresh
+    );
+    if (deleteResponse.isRight() && deleteResponse.value.status === 200) {
+      const notDeletedMethodsCount =
+        deleteResponse.value.value.data?.notDeletedWallets ?? -1;
+      // some error occurred while deletion
+      if (notDeletedMethodsCount !== 0) {
+        yield put(
+          deleteAllPaymentMethodsByFunction.failure({
+            error: Error("can't delete some methods"),
+            notDeletedMethodsCount
+          })
+        );
+        return;
+      }
+      const deletedMethodsCount =
```

Comment:
What an undefined `deletedWallets` means? It's not an error?

Reply:
No it represents the count of wallets deleted
If the request went successfully we expect
`deletedWallets >= 0` && `notDeletedMethodsCount === 0`